### PR TITLE
build(linting): add more lints

### PR DESCRIPTION
**Pull request by Rudxain**
https://rust-lang.github.io/rust-clippy/stable/

### CODE REVIEW

**Comment by AnonymousWP** [comment]
Please rework the title of this PR as it's too generic. Saying this because the changelog content will be based on the title of PRs.

**Comment by Frigyes06** [comment]
Nooo, now we'll have to do proper result handling and such :(
No more bad code

**Comment by Rudxain** [comment]
> Nooo, now we'll have to do proper result handling and such :(

lol 💀😅. We can still use `expect`! My reasoning for `unwrap_used` is that it'll make debugging easier

**Comment by Frigyes06** [comment]
Should the actual fixes to the issues uncovered by the lints be in this PR or a separate PR?

**Comment by Rudxain** [comment]
> Should the actual fixes to the issues uncovered by the lints be in this PR or a separate PR?

I was considering the same thing 🤔

This PR should be for deciding which lints to add and which to `allow` (turn off)

After the decision is made, we can gradually apply the fixes

**Comment by Rudxain** [comment]
Should we `warn` on `pedantic` group? I'm too lazy to find individual lints to enable, lol. So we could simply `allow` individual lints that are annoying, and keep the rest

File: Cargo.toml [hunk 1]
```diff
@@ -51,3 +51,21 @@ strip = "symbols"
...
+unsafe_code = "deny"
+
+[lints.clippy]
+undocumented_unsafe_blocks = "forbid"
+exit = "deny"
+panic_in_result_fn = "warn"
+unwrap_used = "warn"
+infinite_loop = "warn"
+mem_forget = "warn"
+string_to_string = "deny"
```
Comment (Rudxain) [inline Cargo.toml:66]: Should it be downgraded to `warn`? `string.to_string()` is only silly when using an editor/IDE with hover support. For reading on GH, it helps to remember the type of the variable

**Comment by AnonymousWP** [comment]
@Universal-Debloater-Alliance/developers Please pick this up. See comments/questions above.

File: Cargo.toml (hunk 1 above)
Comment (lavafroth) [reply Cargo.toml:66]: `deny` is fine

**Review by lavafroth — APPROVED** [review]

**Comment by Rudxain** [comment]
I'll merge when I have time (probably in ~3 hours) to fix the warnings in a separate PR

**Comment by Rudxain** [comment]
When Rust 1.81 releases on `stable`, we should consider [these lints](https://github.com/rust-lang/rust/pull/120924)

**Pull request merged**